st2actions/st2actions/runners/actionchainrunner.py

```diff
@@ -83,10 +88,15 @@ def _get_rendered_vars(vars):
         context = {SYSTEM_KV_PREFIX: KeyValueLookup()}
         return jinja_utils.render_values(mapping=vars, context=context)
 
-    def get_node(self, node_name=None):
+    def get_node(self, node_name=None, raise_on_failure=False):
+        if not node_name:
+            return None
         for node in self.actionchain.chain:
             if node.name == node_name:
                 return node
+        if raise_on_failure:
+            raise runnerexceptions.ActionRunnerException('Unable to find node with name %s.',
```

Comment:
That's something we could also do in the pre-processing phase (before running the chain), right?